packages/vite/src/node/plugins/worker.ts

```diff
@@ -349,12 +350,6 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
             }
           )
         }
-
-        // TODO: check if this should be removed
-        if (config.isWorker) {
-          s = s.replace('import.meta.url', 'self.location.href')
-          return result()
-        }
```

Comment:
rollup has already replaced `import.meta.url` with corresponding value when we hit here.
Maybe we need to use `/import\.meta\.url/g` instead of `'import.meta.url'` and do this in `transform` or `resolveImportMeta`?
It would work even if `import.meta.url` is preserved when using module worker. But it wouldn't work if we didn't replace `import.meta.url` with `self.location.href` when using classic worker because rollup replaces it with `document.currentScript && document.currentScript.src || document.baseURI`.

Reply:
> Maybe we need to use /import\.meta\.url/g instead of 'import.meta.url' and do this in transform or resolveImportMeta?

use `resolveImportMeta` hook to rewrite it.

Reply:
> Maybe we need to use /import\.meta\.url/g instead of 'import.meta.url' and do this in transform or resolveImportMeta?

use `resolveImportMeta` hook to rewrite it.

Reply:
You mean we should use `resolveImportMeta`?